envoy/upstream/host_description.h

```diff
@@ -202,6 +202,12 @@ class HostDescription {
    *         healthy state via an active healthchecking.
    */
   virtual absl::optional<MonotonicTime> lastHcPassTime() const PURE;
+  virtual MonotonicTime lastTrafficPassTime() const PURE;
+  virtual MonotonicTime lastTrafficPassTime2xx() const PURE;
+  virtual MonotonicTime lastTrafficPassTimeGrpc() const PURE;
+  virtual void setLastTrafficTimeTcpSuccess(MonotonicTime last_traffic_pass_time) const PURE;
+  virtual void setLastTrafficTimeHttp2xx(MonotonicTime last_traffic_pass_time) const PURE;
+  virtual void setLastTrafficTimeGrpcSuccess(MonotonicTime last_traffic_pass_time) const PURE;
```

Comment:
nit: can you refactor this to have a singlet getter and setter that takes an enum type? I think this would be more clear and better for future proofing.

Reply:
fixed.